NitroxClient/GameLogic/Spawning/Metadata/Processor/IOxygenSourceMetadataProcessor.cs

```diff
@@ -0,0 +1,22 @@
+﻿using NitroxClient.GameLogic.Spawning.Metadata.Processor.Abstract;
+using NitroxModel.DataStructures.GameLogic.Entities.Metadata;
+using UnityEngine;
+
+namespace NitroxClient.GameLogic.Spawning.Metadata.Processor;
+
+public class IOxygenSourceMetadataProcessor : EntityMetadataProcessor<IOxygenSourceMetadata>
+{
+    public override void ProcessMetadata(GameObject gameObject, IOxygenSourceMetadata metadata)
+    {
+        IOxygenSource iOxygenSource = gameObject.GetComponent<IOxygenSource>();
+
+        if (iOxygenSource != null)
```

Comment:
```suggestion
        if (gameObject.TryGetComponent(out IOxygenSource iOxygenSource))
```
Makes it a little bit clearer imo. Also you should never compare unity objects against null (see [here](https://discussions.unity.com/t/is-null-returns-false-on-null-objects/821933/11) for explaination), instead threat it like in c++ as a bool